addons/mail/models/mail_thread.py

```diff
@@ -1173,10 +1175,14 @@ def message_route(self, message, message_dict, model=None, thread_id=None, custo
             tools.decode_message_header(message, 'Cc'),
             tools.decode_message_header(message, 'Resent-To'),
             tools.decode_message_header(message, 'Resent-Cc')])
-        rcpt_tos_localparts = [e.split('@')[0].lower() for e in tools.email_split(rcpt_tos)]
+        rcpt_tos_localparts = [
+            e.split('@')[0].lower()
+            for e in tools.email_split(rcpt_tos)
+        ]
+        rcpt_tos_valid_localparts = [to for to in rcpt_tos_localparts]
```

Comment:
This seems more obvious:

```suggestion
        rcpt_tos_valid_localparts = rcpt_tos_localparts.copy()
```